swarm.go

```diff
@@ -97,12 +103,13 @@ type Swarm struct {
 }
 
 // NewSwarm constructs a Swarm
-func NewSwarm(ctx context.Context, local peer.ID, peers peerstore.Peerstore, bwc metrics.Reporter) *Swarm {
+func NewSwarm(ctx context.Context, local peer.ID, peers peerstore.Peerstore, bwc metrics.Reporter, peerLimit int) *Swarm {
```

Comment:
This is probably best done with functional options, so that we don't break existing code.

Reply:
Any idea on how to make that happen ? I couldn't find a swarm 'config' that I could use instead, so I had to add it as an additional argument.

Reply:
You can make the options take the swarm pointer itself.
So:
```
type Option func(s *Swarm) error
func NewSwarm(stuff, opt Option...)
```